backend/app/services/diagnosis_agent.py

```
@@ -0,0 +1,486 @@
+                res_pest = str(item.get("insectKorName", "")).strip()
+                res_crop = str(item.get("cropName", "")).strip()
+                insect_key = str(item.get("insectKey", ""))
+                
+                if res_pest and (official_pest_name in res_pest or res_pest in official_pest_name):
+                    if not fallback_key:
+                        fallback_key = insect_key
+                    if crop and res_crop and crop in res_crop:
+                        best_key = insect_key
+                        break
+        else:
+            root = ET.fromstring(text_resp)
```

Ruff (0.15.10)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)